var.mk

```diff
@@ -1,5 +1,10 @@
 # to be included by Makefile
 
+STACK_EXISTS := $(shell command -v stack 2> /dev/null)
+ifndef STACK_EXISTS
+    $(error "stack is not available please install haskell-stack")
+endif
+
```

Comment:
There is no need, that stack exists, neither for *clean targets nor for the other stuff! As said, setting 'STACK=/bin/true' and `STACK_EXEC=`, i.e. to an empty string should be sufficient.